internal/cache/widget.go

```diff
@@ -24,29 +25,46 @@ func Renderer(wid fyne.Widget) fyne.WidgetRenderer {
 			wid = wd.super()
 		}
 	}
-	renderer, ok := renderers.Load(wid)
+
+	renderersLock.RLock()
+	rinfo, ok := renderers[wid]
+	renderersLock.RUnlock()
 	if !ok {
-		renderer = wid.CreateRenderer()
-		renderers.Store(wid, renderer)
+		rinfo = &rendererInfo{renderer: wid.CreateRenderer()}
+		renderersLock.Lock()
+		renderers[wid] = rinfo
+		renderersLock.Unlock()
 	}
 
-	if renderer == nil {
+	if rinfo == nil {
 		return nil
 	}
-	return renderer.(fyne.WidgetRenderer)
+
+	rinfo.setAlive()
+
+	return rinfo.renderer
 }
 
 // DestroyRenderer frees a render implementation for a widget.
 // This is typically for internal use only.
 func DestroyRenderer(wid fyne.Widget) {
 	Renderer(wid).Destroy()
 
-	renderers.Delete(wid)
+	renderersLock.Lock()
+	delete(renderers, wid)
+	renderersLock.Unlock()
 }
```

Comment:
Should this function also deletes child renderers? 🤔 In that way, if a big widget (which renderer has other widgets) is deleted, all its children will do it too at the same time. Am I missing something?

Reply:
That is an interesting point. It should either be done here or in the driver where they call `DestroyRenderer` I think.

Reply:
Maybe here in `DestroyRenderer()` is the appropriate and easiest way for adding it.

Reply:
Yes, probably true

Reply:
After thinking carefully about this, maybe this isn't necessary as cache.DestroyRenderer is only called when the window is closed, but in that part currently we are cleaning all the objects from the canvas using `cache.ForceCleanFor`. So perhaps this can be revisited later if we find this is really needed

Reply:
That's possible too. We don't make much use of `Destroy` yet. I think this will change in 2.1 if we can get the render / extend smoothed out.

Reply:
Great! So I will mark this as resolved.